packages/marketplace/src/core/private-route.tsx

```diff
@@ -1,36 +1,82 @@
 import * as React from 'react'
 import { Route, RouteProps } from 'react-router'
 import { Redirect } from 'react-router-dom'
-import { connect } from 'react-redux'
-import { ReduxState } from 'src/types/core'
+import { Dispatch } from 'redux'
+import { useSelector, useDispatch } from 'react-redux'
 import RouteFetcher from '../components/hocs/route-fetcher'
-import { LoginType } from '@reapit/cognito-auth'
+import { LoginType, LoginIdentity } from '@reapit/cognito-auth'
+import { selectLoginIdentity, selectLoginType } from '@/selector/auth'
+import { authChangeLoginType } from '@/actions/auth'
+import Routes from '@/constants/routes'
 
-export interface PrivateRouteConnectProps {
-  loginType: LoginType
-}
-
-export interface PrivateRouteProps extends PrivateRouteConnectProps {
+export interface PrivateRouteProps {
   allow: LoginType | LoginType[]
   component: React.FunctionComponent | React.LazyExoticComponent<any>
   exact?: boolean
   fetcher?: boolean
 }
 
-export const PrivateRoute = ({
-  component,
-  allow,
-  fetcher = false,
-  loginType,
-  ...rest
-}: PrivateRouteProps & RouteProps) => {
-  const allowTypes = Array.isArray(allow) ? allow : [allow]
-  allowTypes.includes(loginType)
+export const isNotAllowedToAccess = (loginIdentity?: LoginIdentity) => {
+  if (!loginIdentity) {
+    return false
+  }
+  const { clientId, developerId } = loginIdentity
+  if (!clientId && !developerId) {
+    return true
+  }
+  return false
+}
+
+export const handleChangeLoginType = (
+  loginType: LoginType,
+  allow: LoginType | LoginType[],
+  dispatch: Dispatch,
+  loginIdentity?: LoginIdentity,
+) => {
+  return () => {
+    if (!loginIdentity) {
+      return
+    }
+    if (loginType === 'CLIENT' && allow === 'DEVELOPER' && loginIdentity.developerId) {
+      dispatch(authChangeLoginType('DEVELOPER'))
+    }
+    if (loginType === 'DEVELOPER' && allow === 'CLIENT' && loginIdentity.clientId) {
+      dispatch(authChangeLoginType('CLIENT'))
+    }
+  }
+}
+
+export const handleRedirectToAuthenticationPage = (loginIdentity, allow) => {
+  return () => {
+    if (!loginIdentity) {
+      return
+    }
+    const { clientId, developerId } = loginIdentity
+    if ((allow === 'CLIENT' && !clientId) || (allow === 'DEVELOPER' && !developerId)) {
+      window.location.href = `${Routes.Authentication}/${allow}`
```

Comment:
Can we use [useHistory](https://reacttraining.com/react-router/web/api/Hooks/usehistory) from react-router instead?